kafka-client/src/main/java/dev/responsive/kafka/api/ResponsiveStoreFactory.java

```diff
@@ -0,0 +1,139 @@
+   * {@link TimestampedKeyValueStore} and connect it via the Processor API. If using the DSL, use
+   * {@link #materialized(KeyValueStoreOptions)} instead.
+   *
+   * @param options the {@link KeyValueStoreOptions} to use
+   * @return a store builder that can be used to build a key-value store with the given options
+   *         that uses Responsive's storage for its backend
+   */
+  public static <K, V> StoreBuilder<TimestampedKeyValueStore<K, V>> timestampedKeyValueStoreBuilder(
+      final KeyValueStoreOptions<K, V> options
+  ) {
+    throw new UnsupportedOperationException("Not yet implemented: use ResponsiveDriver for now");
+  }
+
+  /**
+   * Create a {@link StoreBuilder} that can be used to build a Responsive {@link WindowStore}
+   * and connect it via the Processor API. If using the DSL, use
+   * {@link #windowMaterialized(WindowStoreOptions)} instead for most operators and
+   * {@link #streamJoinedStore(String, JoinWindows)} for stream-stream joins specifically.
+   *
+   * @param options the {@link WindowStoreOptions} to use
+   * @return a store builder that can be used to build a window store with the given options
+   *         that uses Responsive's storage for its backend
+   */
+  public static <K, V> StoreBuilder<WindowStore<K, V>> windowStoreBuilder(
+      final WindowStoreOptions<K, V> options
+  ) {
+    throw new UnsupportedOperationException("Not yet implemented: use ResponsiveDriver for now");
+  }
+
+  /**
+   * Create a {@link StoreBuilder} that can be used to build a Responsive
+   * {@link TimestampedWindowStore} and connect it via the Processor API. If using the DSL, use
+   * {@link #windowMaterialized(WindowStoreOptions)} instead for most operators and
+   * {@link #streamJoinedStore(String, JoinWindows)} for stream-stream joins specifically.
+   *
+   * @param options the {@link WindowStoreOptions} to use
+   * @return a store builder that can be used to build a window store with the given options
+   *         that uses Responsive's storage for its backend
+   */
+  public static <K, V> StoreBuilder<TimestampedWindowStore<K, V>> timestampedWindowStoreBuilder(
+      final WindowStoreOptions<K, V> options
+  ) {
+    throw new UnsupportedOperationException("Not yet implemented: use ResponsiveDriver for now");
+  }
+
+  /**
+   * Create a {@link WindowBytesStoreSupplier} for a windowed Responsive state store specific to
+   * stream-stream joins in the DSL. This should be set for both stores in the {@link StreamJoined}
+   * configuration options passed in to the stream-stream join operator.
+   *
+   * @param name the name for this store
+   * @param joinWindows the window configuration for this stream-stream join
+   * @return a store supplier that can be used to build a window store for stream-stream joins
+   *         with the given name that uses Responsive's storage for its backend
+   */
+  public static WindowBytesStoreSupplier streamJoinedStore(
```

Comment:
The main context for this is in my super-long manifesto comment above, but to clarify why this is the only one to return a StoreSupplier, it's because the `StreamJoined` options class in Streams is, awkwardly, the only place raw you can/must use a raw StoreSupplier to plug in a custom store (rather than Materialized or StoreBuilder). I don't remember all the details of why, but basically these have special window stores. Also the parameters are pre-fixed and users currently have to figure out what to plug in themselves, so we may as well provide a nice API where they can just pass in their join definition and let us work out the right values for them